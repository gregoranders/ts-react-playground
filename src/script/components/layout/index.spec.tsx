import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import * as TestSubject from '@components/layout';

describe(`${TestSubject.Layout.displayName}`, () => {
  it('exports default', () => {
    expect(TestSubject.default).toBeDefined();
  });

  const TestComponent = TestSubject.default;

  describe('snapshot', () => {
    it('string', (): void => {
      const snapshot = create(
        <MemoryRouter>
          <TestComponent>test</TestComponent>
        </MemoryRouter>,
      );
      snapshot.toJSON();
      expect(snapshot).toMatchSnapshot();
    });
    it('element', (): void => {
      const snapshot = create(
        <MemoryRouter>
          <TestComponent>
            <span>test</span>
          </TestComponent>
        </MemoryRouter>,
      );
      snapshot.toJSON();
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('mount', () => {
    it('string', () => {
      const testSubject = mount(
        <MemoryRouter>
          <TestComponent>test</TestComponent>
        </MemoryRouter>,
      );
      expect(testSubject.text()).toBe(`HeaderHomeAbouttest© 2020 by Gregor Anders`);
    });
    it('element', () => {
      const testSubject = mount(
        <MemoryRouter>
          <TestComponent>
            <span>test</span>
          </TestComponent>
        </MemoryRouter>,
      );
      expect(testSubject.text()).toBe(`HeaderHomeAbouttest© 2020 by Gregor Anders`);
    });
  });
});
