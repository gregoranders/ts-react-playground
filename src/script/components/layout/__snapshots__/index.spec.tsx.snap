// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot element 1`] = `
Array [
  <header
    className="w3-center"
  >
    <h1
      className="w3-jumbo w3-wide"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Header
      </a>
    </h1>
  </header>,
  <nav
    className="w3-bar w3-theme-d3"
  >
    <a
      aria-current={null}
      className="w3-bar-item w3-button w3-mobile w3-hover-theme"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="w3-bar-item w3-button w3-mobile w3-hover-theme w3-right"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </nav>,
  <main
    className="w3-margin"
  >
    <span>
      test
    </span>
  </main>,
  <footer
    className="w3-padding-32 w3-center"
  >
    © 
    2020
     by Gregor Anders
  </footer>,
]
`;

exports[`Layout snapshot string 1`] = `
Array [
  <header
    className="w3-center"
  >
    <h1
      className="w3-jumbo w3-wide"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Header
      </a>
    </h1>
  </header>,
  <nav
    className="w3-bar w3-theme-d3"
  >
    <a
      aria-current={null}
      className="w3-bar-item w3-button w3-mobile w3-hover-theme"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="w3-bar-item w3-button w3-mobile w3-hover-theme w3-right"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </nav>,
  <main
    className="w3-margin"
  >
    test
  </main>,
  <footer
    className="w3-padding-32 w3-center"
  >
    © 
    2020
     by Gregor Anders
  </footer>,
]
`;
